Mathlib/Analysis/Complex/Hadamard.lean

```diff
@@ -0,0 +1,427 @@
+
+lemma norm_le_interpStrip_of_mem_verticalStrip_zero (hd : DiffContOnCl ℂ f (verticalStrip 0 1))
+    (hB : BddAbove ((norm ∘ f) '' (verticalClosedStrip 0 1))) :
+    ∀ z ∈ verticalStrip 0 1, ‖f z‖ ≤ ‖interpStrip f z‖ := by
+  intro z hz
+  apply tendsto_le_of_eventuallyLE _ _ (eventuallyle f z hB hd hz)
+  · apply tendsto_inf_left
+    simp only [tendsto_const_nhds_iff]
+  -- Proof that we can let epsilon tend to zero.
+  · rw [interpStrip_eq_of_mem_verticalStrip _ _ hz]
+    convert ContinuousWithinAt.tendsto _ using 2
+    · simp only [ofReal_zero, zero_add]
+    · simp_rw [← ofReal_add, norm_eq_abs]
+      have : ∀ x ∈ Ioi 0, (x + sSupNormIm f 0) ^ (1 - z.re) * (x + (sSupNormIm f 1)) ^ z.re
+          = abs (↑(x + sSupNormIm f 0) ^ (1 - z) * ↑(x + sSupNormIm f 1) ^ (z)) := by
+              intro x hx
+              simp only [map_mul]
+              repeat rw [abs_cpow_eq_rpow_re_of_nonneg (le_of_lt (sSupNormIm_eps_pos f hx _)) _]
+              · simp only [sub_re, one_re]
+              · simpa using (ne_comm.mpr (ne_of_lt hz.1))
+              · simpa [sub_eq_zero] using (ne_comm.mpr (ne_of_lt hz.2))
+      apply tendsto_nhdsWithin_congr this _
+      simp only [zero_add]
+      rw [map_mul, abs_cpow_eq_rpow_re_of_nonneg (sSupNormIm_nonneg _ _) _,
+        abs_cpow_eq_rpow_re_of_nonneg (sSupNormIm_nonneg _ _) _]
+      · apply Tendsto.mul
+        · apply Tendsto.rpow_const
+          · nth_rw 2 [← zero_add (sSupNormIm f 0)]
+            exact Tendsto.add_const (sSupNormIm f 0) (tendsto_nhdsWithin_of_tendsto_nhds
+              (Continuous.tendsto continuous_id' _))
+          · right; simp only [sub_nonneg, le_of_lt hz.2]
+        · apply Tendsto.rpow_const
+          · nth_rw 2 [← zero_add (sSupNormIm f 1)]
+            exact Tendsto.add_const (sSupNormIm f 1) (tendsto_nhdsWithin_of_tendsto_nhds
+              (Continuous.tendsto continuous_id' _))
+          · right; simp only [sub_nonneg, le_of_lt hz.1]
+      · simpa using (ne_comm.mpr (ne_of_lt hz.1))
+      · simpa [sub_eq_zero] using (ne_comm.mpr (ne_of_lt hz.2))
+
+/--
+**Hadamard three-line theorem** on `[0,1]`: If `f` is a bounded function, continuous on `[0,1]`
+and differentiable on `(0,1)`, then for `M(x) := sup ((norm ∘ f) '' (re ⁻¹' {x}))` we have that
+`∀ z ∈ [0,1]` the inequality `|f(z)| ≤ |M(0) ^ (1-z)| * |M(1) ^ z|` holds. -/
+lemma norm_le_interpStrip_of_mem_verticalClosedStrip (f : ℂ → E) {z : ℂ}
+    (hz : z ∈ verticalClosedStrip 0 1) (hd : DiffContOnCl ℂ f (verticalStrip 0 1))
+    (hB : BddAbove ((norm ∘ f) '' (verticalClosedStrip 0 1))) :
+    ‖f z‖ ≤ ‖interpStrip f z‖ := by
+  apply le_on_closure (norm_le_interpStrip_of_mem_verticalStrip_zero f hd hB)
+    (Continuous.comp_continuousOn' continuous_norm hd.2)
+    (Continuous.comp_continuousOn' continuous_norm (diffContOnCl_interpStrip f).2)
+  rwa [verticalClosedStrip, ← closure_Ioo zero_ne_one, ← closure_preimage_re] at hz
+
+/--Variant of *Hadamard three-line theorem* on `[0,1]` in simpler terms. -/
```

Comment:
Could you explain the content of the theorem in the docstring, just like you did above?